### PR TITLE
[API] [Homepage] taxons in taxon bar only from menu taxon

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | part of #11250, based on https://github.com/Sylius/Sylius/pull/11291
| License         | MIT

it is waiting for https://github.com/Sylius/Sylius/pull/11291

### CODE REVIEW

File: src/Sylius/Behat/Resources/config/suites/api/homepage/viewing_products.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+                - sylius.behat.context.transform.product
+                - sylius.behat.context.transform.shared_storage
+                - sylius.behat.context.transform.taxon
+
+                - sylius.behat.context.setup.channel
+                - sylius.behat.context.setup.product
+                - sylius.behat.context.setup.taxonomy
+
+                - sylius.behat.context.api.shop.home_page
+                - sylius.behat.context.ui.channel
```
Comment (Tomanhez) [inline src/Sylius/Behat/Resources/config/suites/api/homepage/viewing_products.yml]:
```suggestion
                - sylius.behat.context.ui.channel
```
its needed in api?

File: src/Sylius/Bundle/ApiBundle/SerializerContextBuilder/ChannelContextBuilder.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        $this->channelContext = $channelContext;
+    }
+
+    public function createFromRequest(Request $request, bool $normalization, ?array $extractedAttributes = null): array
+    {
+        $context = $this->decoratedContextBuilder->createFromRequest($request, $normalization, $extractedAttributes);
+        try {
+            $context[ContextKeys::CHANNEL] = $this->channelContext->getChannel();
+        } catch (ChannelNotFoundException $exception) {
+        }
```
Comment (Tomanhez) [inline src/Sylius/Bundle/ApiBundle/SerializerContextBuilder/ChannelContextBuilder.php]:
```suggestion
        } catch (ChannelNotFoundException $exception) {}
```

IMO should be enough

File: src/Sylius/Behat/Context/Api/Shop/HomepageContext.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+namespace Sylius\Behat\Context\Api\Shop;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\ResponseCheckerInterface;
+use Sylius\Behat\Service\SharedStorageInterface;
+use Symfony\Component\HttpFoundation\Request as HttpRequest;
+use Webmozart\Assert\Assert;
+use function Amp\Promise\first;
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Shop/HomepageContext.php]:
```suggestion
```

File: src/Sylius/Behat/Context/Api/Shop/HomepageContext.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    /**
+     * @Then I should not see :firstMenuItem and :secondMenuItem in the menu
+     */
+    public function iShouldNotSeeAndInTheMenu(string ...$unexpectedMenuItems): void
+    {
+        $response = json_decode($this->taxonsClient->getLastResponse()->getContent(), true);
+        $menuItems = array_column($response, 'name');
+
+        foreach ($unexpectedMenuItems as $unexpectedMenuItem) {
+            if (in_array($unexpectedMenuItem, $menuItems)) {
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Shop/HomepageContext.php]:
```suggestion
            if (in_array($unexpectedMenuItem, $menuItems, true)) {
```
Best practice is to always put `true` as a last argument. It is responsible for strict compare of items.

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Product.xml [hunk 5]
```diff
@@ -31,27 +30,55 @@
...
+                    <attribute name="groups">admin:product:create</attribute>
+                </attribute>
+            </collectionOperation>
+
+            <collectionOperation name="get_latest">
+                <attribute name="method">GET</attribute>
+                <attribute name="path">shop/products/latest</attribute>
+                <attribute name="normalization_context">
+                    <attribute name="groups">
+                        <attribute>admin:shop:product:read</attribute>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Product.xml]: We should rather create two separate groups here

File: src/Sylius/Bundle/ApiBundle/DataProvider/LatestProductCollectionDataProvider.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+namespace Sylius\Bundle\ApiBundle\DataProvider;
+
+use ApiPlatform\Core\DataProvider\CollectionDataProviderInterface;
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use Sylius\Bundle\ApiBundle\Serializer\ContextKeys;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Repository\ProductRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+final class LatestProductCollectionDataProvider implements CollectionDataProviderInterface, RestrictedDataProviderInterface
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/DataProvider/LatestProductCollectionDataProvider.php]: You haven't created even one spec for provider and context builder classes :(

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Product.xml (hunk 5 above)
Comment (AdamKasp) [reply src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Product.xml:53]: true, rebase went wrong :(

**Pull request closed without merging**